[threaded-animation-resolution] upload timelines through the remote layer tree transaction

graouts opened the pull request:
#### fc079ac269fc721cf297c0db0bd27e1b44443b49
<pre>
[threaded-animation-resolution] upload timelines through the remote layer tree transaction
<a href="https://bugs.webkit.org/show_bug.cgi?id=300992">https://bugs.webkit.org/show_bug.cgi?id=300992</a>
<a href="https://rdar.apple.com/162873561">rdar://162873561</a>

Reviewed by Matt Woodrow.

We introduced timelines for remote layer tree animations in 301615@main. Currently we create
such timelines, as `RemoteAnimationTimeline` objects, in the UI process exclusively based on
the origin time provided through the remote layer tree transaction.

In this patch we create a system such that timelines in the web process can be sent through
the remote layer tree transaction along with animation effects such that we may express several
timeline types down the line.

To that end, `AnimationTimeline` now exposes an `acceleratedRepresentation()` method that be
overridden by subclasses. Initially, only the `DocumentTimeline` implements an override to create
the monotonic timeline type we currently support in the remote layer tree.

These timelines are then gathered as we gather accelerated animation effects in
`AnimationTimelinesController::updateAcceleratedEffectStacks()` and sent up as a new property
in the remote layer tree transaction.

Upon reception in the UI process, remote animation timelines are created based on this data and,
using a unique identifier, are then associated with the remote animations that are created from
the transaction.

This further paves the way for supporting progress-based timelines.

* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/AcceleratedEffectStackUpdater.cpp:
(WebCore::AcceleratedEffectStackUpdater::updateEffectStacks):
(WebCore::AcceleratedEffectStackUpdater::AcceleratedEffectStackUpdater): Deleted.
* Source/WebCore/animation/AcceleratedEffectStackUpdater.h:
(WebCore::AcceleratedEffectStackUpdater::timelines const):
(WebCore::AcceleratedEffectStackUpdater::timeOrigin const): Deleted.
* Source/WebCore/animation/AnimationTimeline.cpp:
(WebCore::AnimationTimeline::AnimationTimeline):
(WebCore::AnimationTimeline::acceleratedRepresentation):
(WebCore::AnimationTimeline::createAcceleratedRepresentation):
* Source/WebCore/animation/AnimationTimeline.h:
(WebCore::AnimationTimeline::clearAcceleratedRepresentation):
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::AnimationTimelinesController::acceleratedEffectStackUpdater):
(WebCore::AnimationTimelinesController::updateAcceleratedEffectStacks):
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/DocumentTimeline.cpp:
(WebCore::DocumentTimeline::applyPendingAcceleratedAnimations):
(WebCore::DocumentTimeline::createAcceleratedRepresentation):
* Source/WebCore/animation/DocumentTimeline.h:
* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
(WebCore::AcceleratedEffect::create):
(WebCore::AcceleratedEffect::clone const):
(WebCore::AcceleratedEffect::AcceleratedEffect):
* Source/WebCore/platform/animation/AcceleratedEffect.h:
(WebCore::AcceleratedEffect::timeline const):
(WebCore::AcceleratedEffect::timelineIdentifier const):
* Source/WebCore/platform/animation/AcceleratedTimeline.cpp: Added.
(WebCore::AcceleratedTimeline::create):
(WebCore::AcceleratedTimeline::AcceleratedTimeline):
* Source/WebCore/platform/animation/AcceleratedTimeline.h: Added.
(WebCore::AcceleratedTimeline::identifier const):
(WebCore::AcceleratedTimeline::originTime const):
* Source/WebCore/platform/animation/TimelineIdentifier.h: Added.
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateAcceleratedEffectsAndBaseValues):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(conditions_for_header):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::timelines const):
(WebKit::RemoteLayerTreeTransaction::setTimelines):
(WebKit::RemoteLayerTreeTransaction::acceleratedTimelineTimeOrigin const): Deleted.
(WebKit::RemoteLayerTreeTransaction::setAcceleratedTimelineTimeOrigin): Deleted.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimeline.cpp:
(WebKit::RemoteAnimationTimeline::create):
(WebKit::RemoteAnimationTimeline::RemoteAnimationTimeline):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimeline.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimelineRegistry.cpp: Added.
(WebKit::RemoteAnimationTimelineRegistry::update):
(WebKit::RemoteAnimationTimelineRegistry::get const):
(WebKit::RemoteAnimationTimelineRegistry::advanceCurrentTime):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimelineRegistry.h: Added.
(WebKit::RemoteAnimationTimelineRegistry::isEmpty const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::updateTimelineRegistration):
(WebKit::RemoteLayerTreeDrawingAreaProxy::timeline const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
(WebKit::RemoteLayerTreeHost::timeline const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.h:
(WebKit::RemoteScrollingCoordinatorProxy::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxy::timeline const):
(WebKit::RemoteScrollingCoordinatorProxy::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm:
(WebKit::RemoteScrollingCoordinatorProxyIOS::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxyIOS::timeline const):
(WebKit::RemoteScrollingCoordinatorProxyIOS::updateAnimations):
(WebKit::RemoteScrollingCoordinatorProxyIOS::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
(WebKit::RemoteLayerTreeEventDispatcher::updateTimelineRegistration):
(WebKit::RemoteLayerTreeEventDispatcher::timeline):
(WebKit::RemoteLayerTreeEventDispatcher::updateAnimations):
(WebKit::RemoteLayerTreeEventDispatcher::registerTimelineIfNecessary): Deleted.
(WebKit::RemoteLayerTreeEventDispatcher::timeline const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingCoordinatorProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingCoordinatorProxyMac.mm:
(WebKit::RemoteScrollingCoordinatorProxyMac::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxyMac::timeline const):
(WebKit::RemoteScrollingCoordinatorProxyMac::registerTimelineIfNecessary): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::willCommitLayerTree):

Canonical link: <a href="https://commits.webkit.org/301942@main">https://commits.webkit.org/301942@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a97096434dc66f7fc459ff5496af48d35791103d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127564 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47212 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38357 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134641 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79121 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/129436 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47832 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55739 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97105 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/65026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130512 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38257 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114250 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77585 "Passed tests") | | ⏳ 🛠 vision-apple 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/126915 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/37073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/32348 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/78014 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108122 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32787 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137125 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54223 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41784 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105628 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54734 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110608 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105279 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50797 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/29238 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51808 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19947 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/54160 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60247 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53394 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56851 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/55153 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mattwoodrow commented on Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
```diff
@@ -643,19 +643,44 @@ void displayLinkFired(PlatformDisplayID displayID, DisplayUpdate, bool wantsFull
+        if (!foundExistingTimeline) {
+            m_timelines.ensure(processIdentifier, [] {
+                return HashSet<Ref<RemoteAnimationTimeline>>();
+            }).iterator->value.add(RemoteAnimationTimeline::create(timelineRepresentation, now));
```
What removes these timelines when they stop being used?

mattwoodrow replied:
One option here would be to the same as we do for layers. RemoteLayerTreeTransaction has a set of 'changed layers' (which could be added/changed timeline), and a separate set of 'destroyed layers'.

Alternatively, we can just continue to always serialise the entire set, and overwrite the destination. This does need to account for a site-isolation process that can send multiple RemoteLayerTreeTransactions though (and possibly a future where we don't send them all every time).

graouts replied:
Nothing removes them at the moment, this is an oversight. I believe we can and probably should remove them entirely in the remote layer tree once we don't have any animations referencing them.

graouts replied:
Actually, we can remove them during the timeline registration process, since if a timeline becomes unused, it will be missing from the remote layer tree transaction. We will improve upon the remote layer tree transaction data to track more atomic changes in a later patch.

mattwoodrow commented on Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
```diff
@@ -643,19 +643,44 @@ void displayLinkFired(PlatformDisplayID displayID, DisplayUpdate, bool wantsFull
+            auto& existingMonotonicTimelines = it->value;
+            for (auto& existingTimeline : existingMonotonicTimelines) {
+                if (existingTimeline->identifier() == timelineRepresentation->identifier()) {
+                    existingTimeline->setOriginTime(timelineRepresentation->originTime());
```
Do we expect the origin time for a given timeline to be mutable?

graouts replied:
Actually, it can only be set at construction time for a `DocumentTimeline` in the web process, so cannot be updated on the remote side. I'll remove this code and the `RemoteAnimationTimeline::setOriginTime()` method.

mattwoodrow commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimelineRegistry.h:
```diff
@@ -0,0 +1,50 @@
+
+    bool isEmpty() const { return m_timelines.isEmpty(); }
+    void update(WebCore::ProcessIdentifier, const HashSet<Ref<WebCore::AcceleratedTimeline>>&, MonotonicTime);
+    RemoteAnimationTimeline* get(WebCore::ProcessIdentifier, const WebCore::TimelineIdentifier&) const;
```
This code may be simpler if we just did `ProcessQualified<TimelineIdentifer>` (which is mostly the same as passing the pair of arguments). Up to you!

graouts replied:
Oh yes, definitely. I'll either fix it in this patch or land it as a follow-up cleanup. I'll look into making `TimelineIdentifer` process-qualified all the way through as well.

graouts replied:
This is being tracked by [bug 301270](https://bugs.webkit.org/show_bug.cgi?id=301270), I have a patch ready to go as soon as this patch lands.

mattwoodrow commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
```diff
@@ -198,7 +198,7 @@
 
 #if ENABLE(THREADED_ANIMATION_RESOLUTION)
-    Ref { *m_drawingArea }->registerTimelineIfNecessary(processIdentifier, transaction.acceleratedTimelineTimeOrigin(), MonotonicTime::now());
+    Ref { *m_drawingArea }->updateTimelineRegistration(processIdentifier, transaction.timelines(), MonotonicTime::now());
```
A single process can send multiple transactions (with site isolation), which would then overwrite each other in the registry. Please add a TODO/bug for fixing this!

graouts replied:
Will do. Out of curiosity, how should we go about identifying timelines per site-isolated transaction? Is there a permanent identifier that would help there?

graouts replied:
This is tracked by [bug 301261](https://bugs.webkit.org/show_bug.cgi?id=301261).

mattwoodrow replied:
I'm not super sure :)

I think the complexity here is that in the WebProcess, we're collecting timelines on a per-root-frame/transaction granularity, so there can be multiple sets of timelines set at once (in what is as of very recently called a RemoteLayerTreeCommitBundle). In the UI process we're tracking timelines on a per-process/WebPage, so we need to potentially merge the set of timelines from multiple transactions into one registry.

Obviously we could have some sort of begin/end API on the registry, where we can call updateTimelineRegistration multiple times between them, and then clear out the unused entries on end.

I think an alternative would be to change the webprocess/webpage side of the code to accumulate timelines from all the root frames, and store the result on the bundle instead of the transaction. That might be simpler?

The bundle code is super new, and so doesn't really have an example to copy yet.

`RemoteLayerTreeDrawingArea::updateRendering()` currently calls into `webPage->willCommitLayerTree` on a per-transaction basis. I think we're going to need an equivalent that asks the webPage to fill in bundle data (for other reasons, not just this feature), which could then be a spot to accumulate the timelines into the bundle?

graouts committed:
[threaded-animation-resolution] upload timelines through the remote layer tree transaction
https://bugs.webkit.org/show_bug.cgi?id=300992
rdar://162873561

Reviewed by Matt Woodrow.

We introduced timelines for remote layer tree animations in 301615@main. Currently we create
such timelines, as `RemoteAnimationTimeline` objects, in the UI process exclusively based on
the origin time provided through the remote layer tree transaction.

In this patch we create a system such that timelines in the web process can be sent through
the remote layer tree transaction along with animation effects such that we may express several
timeline types down the line.

To that end, `AnimationTimeline` now exposes an `acceleratedRepresentation()` method that be
overridden by subclasses. Initially, only the `DocumentTimeline` implements an override to create
the monotonic timeline type we currently support in the remote layer tree.

These timelines are then gathered as we gather accelerated animation effects in
`AnimationTimelinesController::updateAcceleratedEffectStacks()` and sent up as a new property
in the remote layer tree transaction.

Upon reception in the UI process, remote animation timelines are created based on this data and,
using a unique identifier, are then associated with the remote animations that are created from
the transaction.

This further paves the way for supporting progress-based timelines.

* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/AcceleratedEffectStackUpdater.cpp:
(WebCore::AcceleratedEffectStackUpdater::updateEffectStacks):
(WebCore::AcceleratedEffectStackUpdater::AcceleratedEffectStackUpdater): Deleted.
* Source/WebCore/animation/AcceleratedEffectStackUpdater.h:
(WebCore::AcceleratedEffectStackUpdater::timelines const):
(WebCore::AcceleratedEffectStackUpdater::timeOrigin const): Deleted.
* Source/WebCore/animation/AnimationTimeline.cpp:
(WebCore::AnimationTimeline::AnimationTimeline):
(WebCore::AnimationTimeline::acceleratedRepresentation):
(WebCore::AnimationTimeline::createAcceleratedRepresentation):
* Source/WebCore/animation/AnimationTimeline.h:
(WebCore::AnimationTimeline::clearAcceleratedRepresentation):
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::AnimationTimelinesController::acceleratedEffectStackUpdater):
(WebCore::AnimationTimelinesController::updateAcceleratedEffectStacks):
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/DocumentTimeline.cpp:
(WebCore::DocumentTimeline::applyPendingAcceleratedAnimations):
(WebCore::DocumentTimeline::createAcceleratedRepresentation):
* Source/WebCore/animation/DocumentTimeline.h:
* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
(WebCore::AcceleratedEffect::create):
(WebCore::AcceleratedEffect::clone const):
(WebCore::AcceleratedEffect::AcceleratedEffect):
* Source/WebCore/platform/animation/AcceleratedEffect.h:
(WebCore::AcceleratedEffect::timeline const):
(WebCore::AcceleratedEffect::timelineIdentifier const):
* Source/WebCore/platform/animation/AcceleratedTimeline.cpp: Added.
(WebCore::AcceleratedTimeline::create):
(WebCore::AcceleratedTimeline::AcceleratedTimeline):
* Source/WebCore/platform/animation/AcceleratedTimeline.h: Added.
(WebCore::AcceleratedTimeline::identifier const):
(WebCore::AcceleratedTimeline::originTime const):
* Source/WebCore/platform/animation/TimelineIdentifier.h: Added.
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateAcceleratedEffectsAndBaseValues):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(conditions_for_header):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::timelines const):
(WebKit::RemoteLayerTreeTransaction::setTimelines):
(WebKit::RemoteLayerTreeTransaction::acceleratedTimelineTimeOrigin const): Deleted.
(WebKit::RemoteLayerTreeTransaction::setAcceleratedTimelineTimeOrigin): Deleted.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimeline.cpp:
(WebKit::RemoteAnimationTimeline::create):
(WebKit::RemoteAnimationTimeline::RemoteAnimationTimeline):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimeline.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimelineRegistry.cpp: Added.
(WebKit::RemoteAnimationTimelineRegistry::update):
(WebKit::RemoteAnimationTimelineRegistry::get const):
(WebKit::RemoteAnimationTimelineRegistry::advanceCurrentTime):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteAnimationTimelineRegistry.h: Added.
(WebKit::RemoteAnimationTimelineRegistry::isEmpty const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::updateTimelineRegistration):
(WebKit::RemoteLayerTreeDrawingAreaProxy::timeline const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
(WebKit::RemoteLayerTreeHost::timeline const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.h:
(WebKit::RemoteScrollingCoordinatorProxy::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxy::timeline const):
(WebKit::RemoteScrollingCoordinatorProxy::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm:
(WebKit::RemoteScrollingCoordinatorProxyIOS::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxyIOS::timeline const):
(WebKit::RemoteScrollingCoordinatorProxyIOS::updateAnimations):
(WebKit::RemoteScrollingCoordinatorProxyIOS::registerTimelineIfNecessary): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
(WebKit::RemoteLayerTreeEventDispatcher::updateTimelineRegistration):
(WebKit::RemoteLayerTreeEventDispatcher::timeline):
(WebKit::RemoteLayerTreeEventDispatcher::updateAnimations):
(WebKit::RemoteLayerTreeEventDispatcher::registerTimelineIfNecessary): Deleted.
(WebKit::RemoteLayerTreeEventDispatcher::timeline const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingCoordinatorProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingCoordinatorProxyMac.mm:
(WebKit::RemoteScrollingCoordinatorProxyMac::updateTimelineRegistration):
(WebKit::RemoteScrollingCoordinatorProxyMac::timeline const):
(WebKit::RemoteScrollingCoordinatorProxyMac::registerTimelineIfNecessary): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::willCommitLayerTree):

Canonical link: https://commits.webkit.org/301942@main
